From b6352e70ad1c63b92a0685a956f6ada55109b3e9 Mon Sep 17 00:00:00 2001
From: Biju Chakkachattil <bijux.chakkachattil@intel.com>
Date: Wed, 20 May 2015 11:55:11 +0530
Subject: [PATCH] Change the CHT HR GPIO Pin index to match the BIOS entry.

Latest BIOS add the 3 GPIO pins vbus/vconn/vchrg_in to fix the otg
detection problem in CHT HR. The kernel code GPIO pin entry should
match the index with the BIOS entry. This patch will change the pin
index to match with the BIOS entry.

Change-Id: Icce34abcd4a4b6fa908c31ee2db0b732b05e17db
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-35373
Signed-off-by: Biju Chakkachattil <bijux.chakkachattil@intel.com>
---
 drivers/misc/intel_wcove_gpio.c | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/drivers/misc/intel_wcove_gpio.c b/drivers/misc/intel_wcove_gpio.c
index eeffd13a1782..aef5276a9726 100644
--- a/drivers/misc/intel_wcove_gpio.c
+++ b/drivers/misc/intel_wcove_gpio.c
@@ -171,8 +171,16 @@ static int wcove_gpio_probe(struct platform_device *pdev)
 	 * However ACPI _DSD is not support in Gmin yet and we need to live
 	 * with it.
 	 */
+	info->gpio_vchgrin = devm_gpiod_get_index(&pdev->dev,
+					WCOVE_GPIO_VCHGIN, 0);
+	if (IS_ERR(info->gpio_vchgrin)) {
+		dev_err(&pdev->dev, "Can't request gpio_vchgrin\n");
+		ret = PTR_ERR(info->gpio_vchgrin);
+		goto error_gpio;
+	}
+
 	info->gpio_otg = devm_gpiod_get_index(&pdev->dev,
-					WCOVE_GPIO_OTG, 0);
+					WCOVE_GPIO_OTG, 1);
 	if (IS_ERR(info->gpio_otg)) {
 		dev_err(&pdev->dev, "Can't request gpio_otg\n");
 		ret = PTR_ERR(info->gpio_otg);
@@ -180,18 +188,16 @@ static int wcove_gpio_probe(struct platform_device *pdev)
 	}
 
 	info->gpio_vconn = devm_gpiod_get_index(&pdev->dev,
-					WCOVE_GPIO_VCONN, 1);
+					WCOVE_GPIO_VCONN, 2);
 	if (IS_ERR(info->gpio_vconn)) {
 		dev_err(&pdev->dev, "Can't request gpio_vconn\n");
 		ret = PTR_ERR(info->gpio_vconn);
 		goto error_gpio;
 	}
 
-	info->gpio_vchgrin = devm_gpiod_get_index(&pdev->dev,
-					WCOVE_GPIO_VCHGIN, 2);
-	if (IS_ERR(info->gpio_vchgrin)) {
-		dev_err(&pdev->dev, "Can't request gpio_vchgrin\n");
-		ret = PTR_ERR(info->gpio_vchgrin);
+	ret = gpiod_direction_output(info->gpio_vchgrin, 0);
+	if (ret < 0) {
+		dev_err(&pdev->dev, "cannot configure vchgrin-gpio %d\n", ret);
 		goto error_gpio;
 	}
 
@@ -206,12 +212,6 @@ static int wcove_gpio_probe(struct platform_device *pdev)
 		dev_err(&pdev->dev, "cannot configure vconn-gpio %d\n", ret);
 		goto error_gpio;
 	}
-
-	ret = gpiod_direction_output(info->gpio_vchgrin, 0);
-	if (ret < 0) {
-		dev_err(&pdev->dev, "cannot configure vchgrin-gpio %d\n", ret);
-		goto error_gpio;
-	}
 	dev_dbg(&pdev->dev, "wcove gpio probed\n");
 
 	return 0;
-- 
1.9.1

